### DISCUSSION TITLE
How to handle Uri activation in Avalonia app?

### DISCUSSION THREAD

**Discussion by DmitryBorodiy** [AvaloniaUI/Avalonia#16367] [Q&A] [answered]
Hello, community!
How I can handle Uri activation in Avalonia app?

**Comment by stevemonaco** [comment]
11.1 adds a Launcher that you can get from a `Window` or `TopLevel`. eg.

```cs
public async void ClickHandler(object? sender, RoutedEventArgs e)
{
    if (GetTopLevel(this)?.Launcher is not { } launcher)
        return;

    await launcher.LaunchUriAsync(new Uri("https://avaloniaui.net/"));
}
```

**Comment by maxkatz6** [comment]
See https://docs.avaloniaui.net/docs/concepts/services/activatable-lifetime#handling-uri-activation

**Reply by maxkatz6** [reply, marked as answer]
You still need to setup your platform projects to listen for your URI protocol. 
But on C# side you can use this unified API to listen for events.

**Reply by DmitryBorodiy** [reply]
This API is not supported on Windows(

**Reply by maxkatz6** [reply]
Yea, because Windows doesn't have such concept of "activation". It only can open a new process. You can check by reading `string[] args` from the "Main" method.

**Reply by maxkatz6** [reply]
Uri handling registration on Windows is also more complicated, as you need to modify regedit for that. https://superuser.com/questions/1049803/register-custom-uri-scheme-handler-in-windows-10
